components/org.wso2.carbon.config/src/test/resources/conf/uniqueConfigElement.yaml

```diff
@@ -0,0 +1 @@
+TRANSPORT: NAME
```

Comment:
Please correct me if I'm wrong. AFAIU this is to set the name of the field in the object in an array right? So shouldn't this be also set through an environment variable? 

```
WSO2_TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_UNIQUEFIELD="NAME"
WSO2_TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_NAME="pqr"
WSO2_TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_PORT="${env:pqr.http.port}"
```

Is that going to complicate the array lookup more? Also AFAIU, since these fields are pretty much set when the config beans are written, they wouldn't have to be changed in deployments. So is this going to be generated by the config maven plugin?

Reply:
@chamilad This logic will change. We will initially look for "id" (1st priority), then "name" (2nd priority) and finally as you mentioned via the environment variable itself.

Reply:
+1